### PR TITLE
async refactor of annual IVL renewal coverage initiation

### PR DESCRIPTION
**Pull request by mdkaraman**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: TT6-186390318

# A brief description of the changes

Current behavior:
Beginning coverage for IVL renewal enrollments on 1/1 is a single threaded process.

New behavior:
Begin coverage requests are published by a single threaded process so that individual begin coverage events can be processed in a multi-threaded, asynchronous manner.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
The CodeSweep check is producing a known false positive that matches \# 2 in `CODESWEEP_FALSE_POSITIVES.MD`.

  1. Vulnerability: `Injection.SQL`
  2. Severity: `High`
  3. Cause: Code includes an expression with the `where` keyword.
  4. False positive reasons:
    1. There is no user input being sent to the `.where` query: 
  ```
        current_benefit_period = HbxProfile.current_hbx.benefit_sponsorship.current_benefit_coverage_period
        ivl_enrollments = HbxEnrollment.where(
        :effective_on => { "$gte" => current_benefit_period.start_on, "$lt" => current_benefit_period.end_on },
        :kind.in => ["individual", "coverall"],
        :aasm_state.in => ["auto_renewing", "renewing_coverage_selected"]
      )
```
Note that `current_benefit_period` refers to internally generated data stored on `HbxProfile.current_hbx.benefit_sponsorship.current_benefit_coverage_period`

### CODE REVIEW

File: app/event_source/events/individual/enrollments/begin_coverages/request.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Events
+  module Individual
+    module Enrollments
+      module BeginCoverages
+      # Registers event for initiating start of annual IVL enrollment coverage
+        class  Request < EventSource::Event
```
Comment (saikumar9) [inline app/event_source/events/individual/enrollments/begin_coverages/request.rb:8]:
```suggestion
        class Request < EventSource::Event
```

**Review by ATBull81 — APPROVED** [review]

**Comment by ATBull81** [comment]
👍 looks good pending rebase and rerun of checks

**Review by yosefserkez — APPROVED** [review]
Looks good!

**Review by jacobkagon — APPROVED** [review]
Confirmed that the appscan warning (Ruby SQL injection) is a known false positive listed on the codesweep false positive list.

**Pull request merged**